### DISCUSSION TITLE
Why events does not return selected item

### DISCUSSION THREAD

**Discussion by sovetski** [orchidjs/tom-select#715] [General]
Hi,

I checked this page: https://tom-select.js.org/docs/events/

And for example on `change` event, we have only the `value`, not the complete item object.
In the `item_add`, the `item` parameter returns the HTML tag of the item not the object.

For ex debugging the `value` and `item` parameters looks like:

Object:

```
[
  {
  "value": 584215,
  "text": "Suggestion 5"
  }
]
```

The `on("item_add", function (value, item)` returns:

584215 <div data-value=​"584215" class=​"item" data-ts-item>Suggestion 5</div>​

Is it normal and is it possible to improve it? If we want to get the selected items label (text), how we can do it?

I don't want to do a dirty code by removing the HTML code from item and keep only the text 😅

**Comment by rokasjuz** [comment]
Hi,

Had the same issue. My solution:
```
var ts_instance = document.getElementById('ts-select').tomselect;
ts_instance.on('change', function(){
        var selected_item_text = $(ts_instance.getOption(ts_instance.getValue())).text();
});
```

**Reply by rokasjuz** [reply]
You can get selected item at any point by using 1 and 3 line.

**Reply by sovetski** [reply]
Thanks for your help, it's a shame we have to do some DIY with the code; it should be implemented by default in my opinion

**Reply by rokasjuz** [reply]
One plus that this lib is blazing fast. Code might not be elegant but its still fast :) Also its open source so you can try to contribute and write some additional functions. Every contribution looks good on CV :D
